auth/PhabricatorBMOAuthProvider.php

```diff
@@ -1,5 +1,13 @@
 <?php
 
+try {
```

Comment:
I added this `try/catch` because `docker build -f ./Dockerfile .` bombs with:

```
Step 1/12 : FROM mozilla/mozphab:latest
 ---> db65688039a0
Step 2/12 : ARG EXTENSIONS_PATH=/app/phabricator/src/extensions
 ---> Using cache
 ---> 2ed255053e71
Step 3/12 : COPY differential ${EXTENSIONS_PATH}/differential
 ---> Using cache
 ---> e174e14558cc
Step 4/12 : COPY conduit ${EXTENSIONS_PATH}/conduit
 ---> Using cache
 ---> 609d77a599be
Step 5/12 : COPY auth ${EXTENSIONS_PATH}/auth
 ---> Using cache
 ---> e67eece8ffce
Step 6/12 : COPY auth/auth.css /app/phabricator/webroot/rsrc/css/auth.css
 ---> Using cache
 ---> 1efbd7faa76c
Step 7/12 : COPY auth/auth.js /app/phabricator/webroot/rsrc/js/auth.js
 ---> Using cache
 ---> 9051121aaa54
Step 8/12 : RUN /app/phabricator/bin/celerity map
 ---> Running in 0a9377ea9a04
[2017-06-14 18:17:09] EXCEPTION: (PhabricatorClusterStrandedException) Unable to establish a connection to any database host (while trying "phabricator_config"). All masters and replicas are completely unreachable. at [<phabricator>/src/infrastructure/storage/lisk/PhabricatorLiskDAO.php:140]
arcanist(), phabricator(), phutil()
  #0 PhabricatorLiskDAO::raiseUnreachable(string) called at [<phabricator>/src/infrastructure/storage/lisk/PhabricatorLiskDAO.php:111]
  #1 PhabricatorLiskDAO::newClusterConnection(string, string, string) called at [<phabricator>/src/infrastructure/storage/lisk/PhabricatorLiskDAO.php:66]
  #2 PhabricatorLiskDAO::establishLiveConnection(string) called at [<phabricator>/src/infrastructure/storage/lisk/LiskDAO.php:1008]
  #3 LiskDAO::establishConnection(string) called at [<phabricator>/src/infrastructure/storage/lisk/LiskDAO.php:516]
  #4 LiskDAO::loadRawDataWhere(string, string)
  #5 call_user_func_array(array, array) called at [<phabricator>/src/infrastructure/storage/lisk/LiskDAO.php:476]
  #6 LiskDAO::loadAllWhere(string, string) called at [<phabricator>/src/infrastructure/env/PhabricatorConfigDatabaseSource.php:19]
  #7 PhabricatorConfigDatabaseSource::loadConfig(string) called at [<phabricator>/src/infrastructure/env/PhabricatorConfigDatabaseSource.php:7]
  #8 PhabricatorConfigDatabaseSource::__construct(string) called at [<phabricator>/src/infrastructure/env/PhabricatorEnv.php:249]
  #9 PhabricatorEnv::buildConfigurationSourceStack(boolean) called at [<phabricator>/src/infrastructure/env/PhabricatorEnv.php:95]
  #10 PhabricatorEnv::initializeCommonEnvironment(boolean) called at [<phabricator>/src/infrastructure/env/PhabricatorEnv.php:75]
  #11 PhabricatorEnv::initializeScriptEnvironment(boolean) called at [<phabricator>/scripts/init/lib.php:22]
  #12 init_phabricator_script(array) called at [<phabricator>/scripts/init/init-script.php:10]
  #13 require_once(string) called at [<phabricator>/scripts/__init_script__.php:3]
  #14 require_once(string) called at [<phabricator>/scripts/setup/manage_celerity.php:5]
The command '/bin/sh -c /app/phabricator/bin/celerity map' returned a non-zero code: 255
```